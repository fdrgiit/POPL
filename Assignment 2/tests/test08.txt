a b c d e #
VAR ID COMMA STRING ASSIGN COLON NUM INT DO END GREATER DOT LTEQ #
a -> b c d #
b -> VAR b #
b -> ID COMMA #
b -> STRING c #
c -> c ASSIGN #
c -> COLON #
c -> NUM #
c -> d INT #
d -> DO d #
d -> e END #
e -> GREATER DOT LTEQ # 
##

