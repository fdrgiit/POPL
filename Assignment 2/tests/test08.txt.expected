FIRST(a) = { ID, STRING, VAR }
FIRST(b) = { ID, STRING, VAR }
FIRST(c) = { COLON, DO, GREATER, NUM }
FIRST(d) = { DO, GREATER }
FIRST(e) = { GREATER }
FOLLOW(a) = { $ }
FOLLOW(b) = { COLON, DO, GREATER, NUM }
FOLLOW(c) = { ASSIGN, COLON, DO, GREATER, NUM }
FOLLOW(d) = { $, INT }
FOLLOW(e) = { END }
